src/epd3in7/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! A simple Driver for the Waveshare 3.7" E-Ink Display via SPI
+//!
+//!
+//! Build with the help of documentation/code from [Waveshare](https://www.waveshare.com/wiki/3.7inch_e-Paper_HAT),
+use embedded_hal::{
+    blocking::{delay::DelayMs, spi::Write},
+    digital::v2::{InputPin, OutputPin},
+};
+
+pub(crate) mod command;
+mod constants;
+
+use self::command::Command;
+use self::constants::*;
+
+use crate::buffer_len;
+use crate::color::Color;
+use crate::interface::DisplayInterface;
+use crate::traits::{InternalWiAdditions, RefreshLut, WaveshareDisplay};
+
+/// Width of the display.
+pub const WIDTH: u32 = 280;
+
+/// Height of the display
+pub const HEIGHT: u32 = 480;
+
+/// Default Background Color
+pub const DEFAULT_BACKGROUND_COLOR: Color = Color::White;
+
+const IS_BUSY_LOW: bool = false;
+
+/// Display with Fullsize buffer for use with the 2in9 EPD
+#[cfg(feature = "graphics")]
+pub type Display3in7 =
+    crate::graphics::Display<WIDTH, HEIGHT, false, { WIDTH as usize * HEIGHT as usize / 8 }, Color>;
```

Comment:
As done in https://github.com/caemor/epd-waveshare/pull/128 you should prefer using buffer_len here
` { buffer_len(WIDTH as usize, HEIGHT as usize) }`